pkg/reconciler/revision/resources/deploy.go

```diff
@@ -152,6 +154,10 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingC
 	// If the client provides probes, we should fill in the port for them.
 	rewriteUserProbe(userContainer.LivenessProbe, userPortInt)
 
+	if err := verifySecrets(rev, secretLister); err != nil {
+		return nil, err
```

Comment:
It might make sense to wrap the error to add some context, i.e.

```suggestion
		return nil, fmt.Errorf("failed to verify secrets: %w", err)
```